src/els_completion_provider.erl

```diff
@@ -140,44 +158,66 @@ item_kind_module(Module) ->
    }.
 
 %%==============================================================================
-%% Functions
+%% Functions and Types
 %%==============================================================================
 
--spec functions(els_dt_document:item(), boolean(), boolean()) -> [map()].
-functions(Document, _OnlyExported = false, Arity) ->
-  POIs = els_dt_document:pois(Document, [function]),
-  List = [completion_item_function(POI, Arity) || POI <- POIs],
-  lists:usort(List);
-functions(Document, _OnlyExported = true, Arity) ->
-  Exports   = els_dt_document:pois(Document, [export_entry]),
-  Functions = els_dt_document:pois(Document, [function]),
-  ExportsFA = [FA || #{id := FA} <- Exports],
-  List      = [ completion_item_function(POI, Arity)
-                || #{id := FA} = POI <- Functions, lists:member(FA, ExportsFA)
-              ],
-  lists:usort(List).
-
--spec completion_item_function(poi(), boolean()) -> map().
-completion_item_function(#{id := {F, A}, data := ArgsNames}, false) ->
+-spec functions(els_dt_document:item(), poi_kind(), boolean()) -> [map()].
+functions(Document, POIKind, ExportFormat) ->
+  functions(Document, POIKind, ExportFormat, _ExportedOnly = false).
+
+-spec functions(els_dt_document:item(), poi_kind(), boolean(), boolean()) ->
+  [map()].
+functions(Document, POIKind, ExportFormat, ExportedOnly) ->
+  ExportKind = export_entry_kind(POIKind),
+  ItemKind   = completion_item_kind(POIKind),
+
+  Exports = local_and_included_pois(Document, ExportKind),
+  POIs    = local_and_included_pois(Document, POIKind),
+  FAs     = [FA || #{id := FA} <- Exports],
+  Items   = resolve_functions(POIs, FAs, ExportedOnly, ExportFormat, ItemKind),
+  lists:usort(Items).
+
+-spec function_context(els_dt_document:item(), line(), column()) ->
+  {boolean(), poi_kind()}.
+function_context(Document, Line, Column) ->
```

Comment:
I created this function instead of just passing the arguments `Document, Line, Column` to both `functions` and `exported_functions`, because the context that matters (as returned by `function_context`) is the one on the **current** document and not the one in the document we are fetching the exported functions from.